homeassistant/components/lunatone/config_flow.py

```diff
@@ -0,0 +1,221 @@
+DALI_SCAN_SCHEMA: Final[vol.Schema] = vol.Schema(
+    {
+        vol.Required(
+            CONF_SCAN_METHOD, default=DEFAULT_DALI_DEVICE_SCAN_METHOD
+        ): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=DALI_DEVICE_SCAN_METHODS,
+                translation_key=CONF_SCAN_METHOD,
+                mode=selector.SelectSelectorMode.LIST,
+            ),
+        ),
+    }
+)
+
+
+class LunatoneConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Lunatone config flow."""
+
+    VERSION = 0
+    MINOR_VERSION = 2
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.url: str | None = None
+        self.name: str | None = None
+        self.serial_number: int | None = None
+        self.dali_device_scan_task: asyncio.Task | None = None
+
+    @property
+    def _title(self) -> str:
+        return f"{self.name or 'DALI Gateway'} {self.serial_number}"
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.url = user_input[CONF_URL]
+            data = {CONF_URL: self.url}
+            self._async_abort_entries_match(data)
+            auth = Auth(
+                session=async_get_clientsession(self.hass),
+                base_url=self.url,
+            )
+            info = Info(auth)
+            try:
+                await info.async_update()
+            except aiohttp.InvalidUrlClientError:
+                _LOGGER.debug(("Invalid URL: %s"), self.url)
+                errors["base"] = "invalid_url"
+            except aiohttp.ClientConnectionError:
+                _LOGGER.debug(
+                    (
+                        "Failed to connect to device %s. Check the URL and if the "
+                        "device is connected to power"
+                    ),
+                    self.url,
+                )
+                errors["base"] = "cannot_connect"
+            else:
+                self.name = info.name
+                self.serial_number = info.serial_number
+                await self.async_set_unique_id(str(self.serial_number))
+                if self.source == SOURCE_USER:
+                    self._abort_if_unique_id_configured()
+                    return await self.async_step_dali()
+                self._abort_if_unique_id_mismatch()
+                return self.async_update_reload_and_abort(
+                    self._get_reconfigure_entry(),
+                    data_updates=data,
+                    title=self._title,
+                )
+
+        step_id = "reconfigure"
+        data_schema = RECONFIGURE_SCHEMA
+        if self.source == SOURCE_USER:
+            step_id = "user"
+            data_schema = DATA_SCHEMA
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=data_schema,
```

Comment:
```suggestion
        return self.async_show_form(
            step_id="user",
            data_schema=DATA_SCHEMA,
```

do we need to set step_id to "reconfigure"?

Reply:
I guess not really

Reply:
Done with [28038b5](https://github.com/home-assistant/core/pull/149182/commits/28038b534b8efb106a2d58eb4108b3388937a8ab)